### DISCUSSION TITLE
SingleFile does not overwrite the original when saving an annotated page

### DISCUSSION THREAD

**Discussion by wtannd** [gildas-lormeau/SingleFile#1383] [Q&A]
**Describe the bug**
Even if one sets  in the options: file name conflict resolution -> overwrite the existing file, SingleFile still creates an unique name instead of overwriting it after adding an annotation to an already-saved page.

**To Reproduce**
1. in the options: file name conflict resolution -> overwrite the existing file
2. Save an web page on chrome [ I used format of self-extracting zip (univsersal), e.g., a file named "saved_page (1_14_2024 3_28_01 PM).u.zip.html is created]
3. open the saved page with chrome and click "Annotate the page" from the extension
4. after some changes made, click the "Save" button
5. A unique file is created "saved_page (1_14_2024 3_28_01 PM).u.zip (1).html"

**Expected behavior**
The same saved paged should be overwritten.

**Environment**
 - OS: [Windows 10 Enterprise]
 - Browser: [Chrome]
 - Version: [120.0.6099.217 (Official Build) (64-bit)]

**Additional context**
The issue is the same by using a local profile as suggested in https://github.com/gildas-lormeau/SingleFile/discussions/795

**Comment by gildas-lormeau** [comment]
You should be able to solve this issue by enabling the option `File name > save the filename template data into the page` in the options page of SingleFile.

**Comment by wtannd** [comment]
I did have the option set. I tried it on both linux and windows for firefox and chrome/chromium and the behavior is similar. In either SingleFile save page or annotator editor page, the only possible way to overwrite a file on disk is to set the browser Download option to "Always ask you where to save files". When the browser prompts where to save, you click the file to be overwritten to save.

However, firefox works a little better as the browser prompt shows the exact name to be overwritten while the chrome prompt shows an extra bumped version#, like (1), appended at the end of the file name. If the browser is NOT set to "always ask", then both firefox and chrome will quietly save the file with an unique name (i.e., with a bumped up version#).

**Comment by gildas-lormeau** [comment]
Sorry for the late reply, do you confirm the option `Misc. > save pages in background` is enabled?

**Comment by wtannd** [comment]
Now with the option of save pages in background enabled, everything works in both firefox and chrome.

**Comment by gildas-lormeau** [comment]
Thanks for your feedback, I'm glad to know that was the cause of the problem.
